### PR TITLE
Bytecode tests for static methods

### PR DESCRIPTION
**Pull request by AlexeyBarabash**
Resolves https://github.com/brave/brave-browser/issues/38401

This PR expands `BytecodeTest.methodExists` test to detect change of method modifier.
Also this PR removes unnecessary tests against `Brave***` classes, because we own that code and it will be changed by us if we need after the upstream changes.

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
`npm run test --  brave_unit_tests  Release  --target_os=android --target_arch=x86`

### CODE REVIEW

File: android/javatests/org/chromium/chrome/browser/BytecodeTest.java [hunk 1]
```diff
@@ -407,218 +413,360 @@ public void testClassesExist() throws Exception {
...
+        Assert.assertTrue(
+                methodExists(
+                        "org/chromium/chrome/browser/ui/"
+                                + "default_browser_promo/DefaultBrowserPromoUtils",
+                        "prepareLaunchPromoIfNeeded",
+                        MethodModifier.STATIC,
+                        false,
+                        null));
+
+        // Why do we have this below? we can guarantee whatever we need at Brave code
```
Comment (AlexeyBarabash) [inline android/javatests/org/chromium/chrome/browser/BytecodeTest.java:515]: @samartnik , I forgot why do we need the `methodExists` tests for `Brave*****` classes. This looks suspicious here and below in several places. Can you please remind me why we did so?

Comment (samartnik) [reply android/javatests/org/chromium/chrome/browser/BytecodeTest.java:515]: yes, this seems to be overkill, the main point of these tests is to detect changes in the upstream

Comment (AlexeyBarabash) [reply android/javatests/org/chromium/chrome/browser/BytecodeTest.java:515]: thanks for the confirmation @samartnik , I am updating the PR.

**Review by samartnik — APPROVED** [review]
lgtm

**Review by deeppandya — APPROVED** [review]
LGTM

**Review by SergeyZhukovsky — APPROVED** [review]
++

**Pull request merged**